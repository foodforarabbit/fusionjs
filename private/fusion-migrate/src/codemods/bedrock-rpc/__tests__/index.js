const fs = require('fs');
const path = require('path');
const pluginTester = require('babel-plugin-tester');

const tests = fs
  .readdirSync(path.join(__dirname, '../__fixtures__'))
  .map(fixture => {
    return {
      fixture: path.join(__dirname, '../__fixtures__', fixture, 'code.js'),
    };
  });

pluginTester({
  plugin: require('../plugin'),
  snapshot: true,
  tests,
});
