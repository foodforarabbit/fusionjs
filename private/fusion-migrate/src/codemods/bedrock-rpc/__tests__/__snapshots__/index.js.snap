// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bedrock-rpc anonymous-function: anonymous-function 1`] = `
"import RPC from '@uber/bedrock/web-rpc';
import M3 from '@uber/bedrock/universal-m3';
import Logger from '@uber/bedrock/universal-logger';
import WebRpcAtreyu from '@uber/web-rpc-atreyu';
import getAuthParams from '@uber/get-auth-params';
import {rpcIds} from '../shared/rpc-constants';
import atreyuGraphs from './atreyu/graphs';
import atreyuSeeds from './atreyu/seeds';
import atreyuTransforms from './atreyu/transforms';

export default function(server) {
  const webRpcAtreyu = new WebRpcAtreyu(
    server.clients.atreyu,
    server.config.webRpcAtreyu
  );

  const methods = {
    [rpcIds.getUser]: webRpcAtreyu({
      graphDef: atreyuGraphs.user,
      seed: atreyuSeeds.user,
      transform: atreyuTransforms.user,
    }),
  };

  RPC.lol('test');

  return RPC.init(methods, M3, Logger.createChild('web-rpc'));
}


      ↓ ↓ ↓ ↓ ↓ ↓

import {BedrockCompatToken} from '@uber/fusion-plugin-bedrock-compat';
import {createPlugin} from 'fusion-core';
import M3 from '@uber/bedrock/universal-m3';
import Logger from '@uber/bedrock/universal-logger';
import WebRpcAtreyu from '@uber/web-rpc-atreyu';
import getAuthParams from '@uber/get-auth-params';
import {rpcIds} from '../shared/rpc-constants';
import atreyuGraphs from './atreyu/graphs';
import atreyuSeeds from './atreyu/seeds';
import atreyuTransforms from './atreyu/transforms';

function getRPCHandlers(server) {
  const webRpcAtreyu = new WebRpcAtreyu(
    server.clients.atreyu,
    server.config.webRpcAtreyu
  );
  const methods = {
    [rpcIds.getUser]: webRpcAtreyu({
      graphDef: atreyuGraphs.user,
      seed: atreyuSeeds.user,
      transform: atreyuTransforms.user,
    }),
  };
  RPC.lol('test');
  Object.keys(methods).forEach(method => {
    const oldMethod = methods[method];

    methods[method] = (args, ctx) => {
      const req = ctx.req;
      req.body = args;
      return new Promise((resolve, reject) => {
        return oldMethod(req, resolve, reject);
      });
    };
  });
  return methods;
}

export default createPlugin({
  deps: {
    server: BedrockCompatToken,
  },
  provides: ({server}) => getRPCHandlers(server),
});
"
`;

exports[`bedrock-rpc arrow-function: arrow-function 1`] = `
"import RPC from '@uber/bedrock/web-rpc';
import M3 from '@uber/bedrock/universal-m3';
import Logger from '@uber/bedrock/universal-logger';
import WebRpcAtreyu from '@uber/web-rpc-atreyu';
import getAuthParams from '@uber/get-auth-params';
import {rpcIds} from '../shared/rpc-constants';
import atreyuGraphs from './atreyu/graphs';
import atreyuSeeds from './atreyu/seeds';
import atreyuTransforms from './atreyu/transforms';

export default server => {
  const webRpcAtreyu = new WebRpcAtreyu(
    server.clients.atreyu,
    server.config.webRpcAtreyu
  );

  const methods = {
    [rpcIds.getUser]: webRpcAtreyu({
      graphDef: atreyuGraphs.user,
      seed: atreyuSeeds.user,
      transform: atreyuTransforms.user,
    }),
  };

  return RPC.init(methods, M3, Logger.createChild('web-rpc'));
};


      ↓ ↓ ↓ ↓ ↓ ↓

import {BedrockCompatToken} from '@uber/fusion-plugin-bedrock-compat';
import {createPlugin} from 'fusion-core';
import M3 from '@uber/bedrock/universal-m3';
import Logger from '@uber/bedrock/universal-logger';
import WebRpcAtreyu from '@uber/web-rpc-atreyu';
import getAuthParams from '@uber/get-auth-params';
import {rpcIds} from '../shared/rpc-constants';
import atreyuGraphs from './atreyu/graphs';
import atreyuSeeds from './atreyu/seeds';
import atreyuTransforms from './atreyu/transforms';

const getRPCHandlers = server => {
  const webRpcAtreyu = new WebRpcAtreyu(
    server.clients.atreyu,
    server.config.webRpcAtreyu
  );
  const methods = {
    [rpcIds.getUser]: webRpcAtreyu({
      graphDef: atreyuGraphs.user,
      seed: atreyuSeeds.user,
      transform: atreyuTransforms.user,
    }),
  };
  Object.keys(methods).forEach(method => {
    const oldMethod = methods[method];

    methods[method] = (args, ctx) => {
      const req = ctx.req;
      req.body = args;
      return new Promise((resolve, reject) => {
        return oldMethod(req, resolve, reject);
      });
    };
  });
  return methods;
};

export default createPlugin({
  deps: {
    server: BedrockCompatToken,
  },
  provides: ({server}) => getRPCHandlers(server),
});
"
`;

exports[`bedrock-rpc function-declaration: function-declaration 1`] = `
"import RPC from '@uber/bedrock/web-rpc';
import M3 from '@uber/bedrock/universal-m3';
import Logger from '@uber/bedrock/universal-logger';
import WebRpcAtreyu from '@uber/web-rpc-atreyu';
import getAuthParams from '@uber/get-auth-params';
import {rpcIds} from '../shared/rpc-constants';
import atreyuGraphs from './atreyu/graphs';
import atreyuSeeds from './atreyu/seeds';
import atreyuTransforms from './atreyu/transforms';

export default function rpc(server) {
  const webRpcAtreyu = new WebRpcAtreyu(
    server.clients.atreyu,
    server.config.webRpcAtreyu
  );

  const methods = {
    [rpcIds.getUser]: webRpcAtreyu({
      graphDef: atreyuGraphs.user,
      seed: atreyuSeeds.user,
      transform: atreyuTransforms.user,
    }),
  };

  return RPC.init(methods, M3, Logger.createChild('web-rpc'));
}


      ↓ ↓ ↓ ↓ ↓ ↓

import {BedrockCompatToken} from '@uber/fusion-plugin-bedrock-compat';
import {createPlugin} from 'fusion-core';
import M3 from '@uber/bedrock/universal-m3';
import Logger from '@uber/bedrock/universal-logger';
import WebRpcAtreyu from '@uber/web-rpc-atreyu';
import getAuthParams from '@uber/get-auth-params';
import {rpcIds} from '../shared/rpc-constants';
import atreyuGraphs from './atreyu/graphs';
import atreyuSeeds from './atreyu/seeds';
import atreyuTransforms from './atreyu/transforms';

function rpc(server) {
  const webRpcAtreyu = new WebRpcAtreyu(
    server.clients.atreyu,
    server.config.webRpcAtreyu
  );
  const methods = {
    [rpcIds.getUser]: webRpcAtreyu({
      graphDef: atreyuGraphs.user,
      seed: atreyuSeeds.user,
      transform: atreyuTransforms.user,
    }),
  };
  Object.keys(methods).forEach(method => {
    const oldMethod = methods[method];

    methods[method] = (args, ctx) => {
      const req = ctx.req;
      req.body = args;
      return new Promise((resolve, reject) => {
        return oldMethod(req, resolve, reject);
      });
    };
  });
  return methods;
}

export default createPlugin({
  deps: {
    server: BedrockCompatToken,
  },
  provides: ({server}) => rpc(server),
});
"
`;
